File: Python/Clothes/Clothe/Application/CreateClothe/CreateClotheCommandHandler.py
from Clothe.Application.Result import Result
from Clothe.Application.CreateClothe.CreateClotheCommandValidator import CreateClotheCommandValidator
from Clothe.Domain.Entities.ClotheEntity import ClotheEntity
from Clothe.Domain.Repositories import IRepository
from Clothe.Infra.Repositories import ClotheRepository


class CreateClotheCommandHandler:
    def __init__(self, validator: CreateClotheCommandValidator, repository: ClotheRepository):
        self._validator = validator
        self._repository = repository

    def handle(self, request):
        validation_errors = self._validator.validate(request)

        if validation_errors:
            return Result(success=False, message="Invalid data", data=validation_errors)

        clothe = ClotheEntity(request.name, request.model, request.year, request.size, request.color)
        self._repository.add_async(clothe)
        self._repository.save_changes_async()

        return Result(success=True, message="Clothe created successfully", data=clothe.id)

File: .Net/UnitTest/src/Clothe.Application/CreateClothe/CreateClotheCommandHandler.cs
using MediatR;
using Clothes.Application.Results;
using Clothes.Domain.Entities;
using Clothes.Domain.Repositories;

namespace Clothes.Application.CreateClothe;

public class CreateClotheCommandHandler(CreateClotheCommandValidator _validator, IRepository<ClotheEntity> _repository) : IRequestHandler<CreateClotheCommand, Result>
{
    
    public async Task<Result> Handle(CreateClotheCommand request, CancellationToken cancellationToken)
    {
        var isValid = await _validator.ValidateAsync(request, cancellationToken);
        if (!isValid.IsValid)
        {
            return new Result(false, "Invalid data", isValid.Errors);
        }

        var Clothe = new ClotheEntity(request.Name, request.Model, request.Year, request.Size, request.Color);
        await _repository.AddAsync(Clothe, cancellationToken);
        await _repository.SaveChangesAsync(cancellationToken);

        return new Result(true, "Clothe created successfully", Clothe.Id);

    }
}
